src/config/config_impl.go

```diff
@@ -112,19 +115,21 @@ func (this *rateLimitDescriptor) loadDescriptors(config RateLimitConfigToLoad, p
 		var rateLimit *RateLimit = nil
 		var rateLimitDebugString string = ""
 		if descriptorConfig.RateLimit != nil {
+			unlimited := descriptorConfig.RateLimit.Unlimited
+
 			value, present :=
 				pb.RateLimitResponse_RateLimit_Unit_value[strings.ToUpper(descriptorConfig.RateLimit.Unit)]
-			if !present || value == int32(pb.RateLimitResponse_RateLimit_UNKNOWN) {
+			if (!present || value == int32(pb.RateLimitResponse_RateLimit_UNKNOWN)) && !unlimited {
```

Comment:
Can you split this logic out so that we actually check if unlimited is set with a unit at all and have an error message (and test) for that? I think that would be more clear to the user.

Reply:
(Ping on this)

Reply:
I wasn't too sure what exactly you meant, so please review this again :)